src/pages/api/progress.ts

```diff
@@ -0,0 +1,70 @@
+import TinCan from 'tincanjs'
+import {NextApiRequest, NextApiResponse} from 'next'
+import fetchEggheadUser from 'api/egghead/users/from-token'
+import {getTokenFromCookieHeaders} from 'utils/auth'
+
+const SCORM_CLOUD_ENDPOINT = `https://cloud.scorm.com/lrs/PZ9CYEKKV8/`
+
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+  const {eggheadToken} = getTokenFromCookieHeaders(req.headers.cookie as string)
+
+  if (req.method === 'POST') {
+    try {
+      const lrs = new TinCan.LRS({
+        endpoint: SCORM_CLOUD_ENDPOINT,
+        username: process.env.SCORM_CLOUD_USERNAME,
+        password: process.env.SCORM_CLOUD_PASSWORD,
+        allowFail: false,
+      })
+
+      const eggheadViewer = await fetchEggheadUser(eggheadToken, true)
+
+      const {verb, target} = req.body
+
+      const statement: any = new TinCan.Statement({
+        actor: {
+          account: {
+            mbox: `mailto:${eggheadViewer.email}`,
+            homePage: `https://egghead.io/users/${eggheadViewer.id}`,
+            name: `${eggheadViewer.id}`,
+          },
+        },
+        verb: {
+          id: `https://w3id.org/xapi/dod-isd/verbs/${verb}`,
+          display: {
+            'en-US': verb,
+          },
+        },
+        target: target,
+      })
+
+      const result: any = await new Promise((resolve, reject) => {
```

Comment:
creating a promise because the `tincanjs` uses some format of low level XHR request that I don't understand at all. 😅

Reply:
this is the kind of shit that makes people quit programming computers.